woodwork/tests/schema/test_schema.py

```diff
@@ -309,6 +310,141 @@ def test_get_subset_schema_all_params(sample_column_names, sample_inferred_logic
     assert schema is not copy_schema
 
 
+def test_set_logical_types(sample_column_names, sample_inferred_logical_types):
+    semantic_tags = {
+        'full_name': 'tag1',
+        'email': ['tag2'],
+        'phone_number': ['tag3', 'tag2'],
+        'signup_date': {'secondary_time_index'},
+    }
+    schema = Schema(sample_column_names, sample_inferred_logical_types,
+                    semantic_tags=semantic_tags, use_standard_tags=True)
+
+    schema.set_types(logical_types={
+        'full_name': Categorical,
+        'email': EmailAddress,
+        'phone_number': PhoneNumber,
+        'age': Double,
+    })
+
+    assert schema.logical_types['full_name'] == Categorical
+    assert schema.logical_types['email'] == EmailAddress
+    assert schema.logical_types['phone_number'] == PhoneNumber
+    assert schema.logical_types['age'] == Double
+
+    # Verify semantic tags were reset to standard tags in columns with Logical Type changes
+    assert schema.semantic_tags['full_name'] == {'category'}
+    assert schema.semantic_tags['email'] == set()
+    assert schema.semantic_tags['phone_number'] == set()
+    assert schema.semantic_tags['age'] == {'numeric'}
+
+    # Verify signup date column was unchanged
+    assert schema.logical_types['signup_date'] == Datetime
+    assert schema.semantic_tags['signup_date'] == {'secondary_time_index'}
+
+
+def test_set_logical_types_empty(sample_column_names, sample_inferred_logical_types):
+    semantic_tags = {
+        'full_name': 'tag1',
+        'age': 'test_tag'
+    }
+    schema = Schema(sample_column_names, sample_inferred_logical_types,
+                    index='full_name',
+                    semantic_tags=semantic_tags, use_standard_tags=True)
+
+    # If the tags for a column are None, nothing should change
+    schema.set_types(semantic_tags={'full_name': None}, retain_index_tags=False)
```

Comment:
I don't know why a user would do this, but figured it'd be better to just do nothing than to error or still reset. Happy to discuss other options, though

Reply:
I think I'd like to catch the None case in `_check_semantic_tags`--that way it'd also get caught at init.

Reply:
Makes sense to me